presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushPredicateIntoTableScan.java

```diff
@@ -149,6 +149,10 @@ private boolean arePlansSame(FilterNode filter, TableScanNode tableScan, PlanNod
         // don't include non-deterministic predicates
         Expression deterministicPredicate = filterDeterministicConjuncts(predicate);
 
+        if (TRUE_LITERAL.equals(deterministicPredicate)) {
+            return Optional.empty();
```

Comment:
You should also return when `newDomain` is `all` and `!pruneWithPredicateExpression`. I.e. when `!pruneWithPredicateExpression` and there is some deterministic predicate that cannot be converted to a tuple domain (e.g an OR, or a function call).

TL;DR https://github.com/prestosql/presto/pull/1312